src/main/scala/edu/ie3/simona/model/participant/HpModel.scala

```diff
@@ -116,19 +118,23 @@ final case class HpModel(
     * function calculates the heat pump's next state to get the actual active
     * power of this state use [[calculateActivePower]] with the generated state
     *
-    * @param state
-    *   Current state of the heat pump
+    * @param lastState
+    *   Last state of the heat pump
     * @param relevantData
     *   data of heat pump including
     * @return
-    *   next [[HpState]]
+    *   Booleans if Hp can operate and can be out of operation plus next
+    *   [[HpState]]
     */
   def determineState(
-      state: HpState,
+      lastState: HpState,
       relevantData: HpRelevantData,
-  ): HpState = {
-    val turnOn = operatesInNextState(state, relevantData)
-    calcState(state, relevantData, turnOn)
+  ): (Boolean, Boolean, HpState) = {
```

Comment:
You do not want them to be included in 'operatesInNextState'? This would require passing `demandHouse`, `demandThermalStorage` plus 'noThermalStorageOrThermalStorageIsEmpty' (and the demand for hotWaterStorage, which we might introduce later with another PR) into these separate methods. To me, this looks more complicated at first glance. But maybe you have a good idea?

Reply:
Made a proposal, maybe it suites your whishes. Let me know, what you're thinking.

Reply:
In think that's fine for now.